besu/src/main/java/org/hyperledger/besu/cli/util/VersionProvider.java

```diff
@@ -15,12 +15,27 @@
 package org.hyperledger.besu.cli.util;
 
 import org.hyperledger.besu.BesuInfo;
+import org.hyperledger.besu.services.PluginVersionsProvider;
+
+import java.util.ArrayList;
+import java.util.List;
 
 import picocli.CommandLine;
 
 public class VersionProvider implements CommandLine.IVersionProvider {
+  private final PluginVersionsProvider pluginVersionsProvider;
+
+  public VersionProvider(final PluginVersionsProvider pluginVersionsProvider) {
+    this.pluginVersionsProvider = pluginVersionsProvider;
+  }
+
   @Override
   public String[] getVersion() {
-    return new String[] {BesuInfo.version()};
+    // the PluginVersionsProvider has registered plugins and their versions by this time.
+    final List<String> pluginVersionsList = this.pluginVersionsProvider.getPluginVersions();
```

Comment:
nit: pluginVersions instead of pluginVersionsList and same with versions below

Reply:
@jframe I've simplified the conversion to array using Stream, please re-review.